native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -0,0 +1,257 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  StyleSheet,
+  TouchableNativeFeedback,
+  TouchableWithoutFeedback,
+  View,
+  ViewPropTypes,
+} from 'react-native';
+
+import {
+  borderRadiusSm,
+  colorBlue500,
+  colorGray50,
+  colorGray500,
+  colorGreen500,
+  colorRed500,
+  colorYellow500,
+  spacingBase,
+  spacingLg,
+  spacingMd,
+  spacingSm,
+  spacingXxl,
+} from 'bpk-tokens/tokens/base.react.native';
+
+import BpkAnimateHeight from 'react-native-bpk-component-animate-height';
+import BpkIcon from 'react-native-bpk-component-icon';
+import BpkText from 'react-native-bpk-component-text';
+
+import AnimateAndFade from './AnimateAndFade';
+import {
+  dismissablePropType,
+  toggleExpandedButtonLabelPropType,
+  dismissableLabelPropType,
+} from './customPropTypes';
+
+export const ALERT_TYPES = {
+  SUCCESS: 'success',
+  WARN: 'warn',
+  ERROR: 'error',
+  NEUTRAL: 'neutral',
+};
+
+const styles = StyleSheet.create({
+  mainContainer: {
+    paddingVertical: spacingBase,
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+    minHeight: spacingXxl + spacingBase,
+    flex: 1,
+  },
+  dismissableContainer: {
+    paddingHorizontal: spacingLg - spacingMd,
+    backgroundColor: colorGray50,
+    justifyContent: 'center',
+  },
+  childrenContainer: {
+    paddingVertical: spacingSm,
+  },
+  flexRow: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  text: {
+    flex: 1,
+  },
+  icon: {
+    paddingRight: spacingSm,
+    paddingTop: spacingSm - 1,
+  },
+  dismissLabel: {
+    color: colorBlue500,
+  },
+  dismissButtonContainer: {
+    padding: spacingMd,
+    borderRadius: borderRadiusSm,
+  },
+  dismissButtonContainerRadius: {
+    borderRadius: borderRadiusSm,
+  },
+});
+
+const ALERT_TYPE_STYLES = {
+  [ALERT_TYPES.SUCCESS]: {
+    iconSource: 'tick-circle',
+    iconStyle: {
+      color: colorGreen500,
+    },
+  },
+  [ALERT_TYPES.WARN]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorYellow500,
+    },
+  },
+  [ALERT_TYPES.ERROR]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorRed500,
+    },
+  },
+  [ALERT_TYPES.NEUTRAL]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorGray500,
+    },
+  },
+};
+
+const dismissableButton = (dismissable, dismissButtonLabel, onDismiss) => {
```

Comment:
My gut feeling says this would be better written as a React component and not a function, e.g.:

```
const DismissableButton = ({ dismissButtonLabel, onDismiss }) => (
  <View style={styles.dismissableContainer}>
     ...
  </View>
);
```

So then it could be called from `BpkBannerAlert` like this:

```
{dismissable && <DismissableButton dismissButtonLabel={dismissButtonLabel} onDismiss={onDismiss} />}
```
as opposed to:
```
{dismissable && dismissableButton(dismissable, dismissButtonLabel, onDismiss)}
```

But in fairness, I'm unsure if there are any specific advantages we get from changing it to this. (any thoughts @matthewdavidson @k0nserv?)

(same applies to `expandableContent` and `expandableIcon` too)